Framework/Algorithms/src/FitPeaks.cpp

```diff
@@ -2010,17 +2011,8 @@ void FitPeaks::generateCalculatedPeaksWS() {
     return;
   }
 
-  // create a wokspace with same number of input matrix workspace
+  // create a wokspace with same size as in the input matrix workspace
   m_fittedPeakWS = create<Workspace2D>(*m_inputMatrixWS);
-  for (size_t iws = 0; iws < m_fittedPeakWS->getNumberHistograms(); ++iws) {
```

Comment:
Just to confirm that the X values are set somewhere else so this loop is no longer needed?

Reply:
Yes, `create<>` copies them from the input if it has the same size as the parent, see:

https://github.com/mantidproject/mantid/blob/master/Framework/DataObjects/inc/MantidDataObjects/WorkspaceCreation.h#L209-L217

And a test:

https://github.com/mantidproject/mantid/blob/master/Framework/DataObjects/test/WorkspaceCreationTest.h#L311-L325

Reply:
Thanks. I'd forgotten that part of the new creation methods.